common.py

```diff
@@ -173,12 +190,6 @@ def from_dict(cls: Type[T], data: Dict[str, Any]) -> T:
         field_type = field.type
         if is_dataclass(field_type) and field_name in data:
             field_values[field_name] = from_dict(field_type, data[field_name])
-        elif (
-            isinstance(field.type, type)
-            and issubclass(field.type, Enum)
-            and field_name in data
-        ):
-            field_values[field_name] = enum_factory(field.type)(data[field_name])
         elif field_name in data:
             field_values[field_name] = data[field_name]
-    return cls(**field_values)
+    return cast(T, cls(**field_values))
```

Comment:
this function is was introduced by the branch, and modified twice afterwards (on the same branch). Can this part of the branch be rebased?